### PR TITLE
Remember caption option in player, closes #2811

### LINKED ISSUE #2811: Remember caption option in player
- [✔️] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
- [✔️] I checked if the issue/feature exists in the latest version.
- [I don't have json bugreport since this is feature request so ❌] I did use the [incredible bugreport to markdown converter](https://teamnewpipe.github.io/CrashReportToMarkdown/) to paste bug reports.

Currently it forgets and always turn off by default. Quite annoying because i have to turn on again and again
Can you make it remember when i turned it on and remember language? Doesn't matter if it's english or english (auto generated)

### PR DESCRIPTION
**Pull request by Redirion**
There had been feature requests like #1329 and #2392.
But instead of adding a new preference which seems to be generally not favourable, the request in #2811 seems actually pretty straight forward.
This PR implements it.

I have also considered the auto-generated subtitles (and hopefully internationalized, I have tested english and german phone languages). If auto-generated language was chosen last time by the user, the non auto-generated language will be preferred, but if the native chosen language is not available, the auto-generated variant will be used as a fallback.

Debug apk for testing: https://github.com/Redirion/NewPipe/releases/tag/v0.17.4cc

- [X] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.

### CODE REVIEW

File: app/src/main/java/org/schabi/newpipe/player/VideoPlayer.java [hunk 1]
```diff
@@ -285,6 +288,12 @@ private void buildCaptionMenu(final List<String> availableLanguages) {
         if (captionPopupMenu == null) return;
         captionPopupMenu.getMenu().removeGroup(captionPopupMenuGroupId);
 
+        String userPreferredLanguage = PreferenceManager.getDefaultSharedPreferences(context)
+                .getString(context.getString(R.string.caption_user_set_key), null);
+        // english (auto-generated)
+        boolean searchForAutogenerated = userPreferredLanguage == null ||
+                !userPreferredLanguage.contains("(");
```
Comment (TobiGr) [inline app/src/main/java/org/schabi/newpipe/player/VideoPlayer.java]: What does `!userPreferredLanguage.contains("(")` do? Please add a comment to ensure people can understand your code later.

File: app/src/main/java/org/schabi/newpipe/player/VideoPlayer.java [hunk 2]
```diff
@@ -308,9 +319,26 @@ private void buildCaptionMenu(final List<String> availableLanguages) {
...
                     trackSelector.setParameters(trackSelector.buildUponParameters()
                             .setRendererDisabled(textRendererIndex, false));
+                    final SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(context);
+                    prefs.edit().putString(context.getString(R.string.caption_user_set_key),
+                            captionLanguage).commit();
                 }
                 return true;
             });
+            // apply caption language from previous user preference
+            if(userPreferredLanguage != null && (captionLanguage.equals(userPreferredLanguage) ||
```
Comment (TobiGr) [inline app/src/main/java/org/schabi/newpipe/player/VideoPlayer.java]:
```suggestion
            if (userPreferredLanguage != null && (captionLanguage.equals(userPreferredLanguage) ||
```

File: app/src/main/java/org/schabi/newpipe/player/VideoPlayer.java (hunk 1 above)
Comment (Redirion) [reply app/src/main/java/org/schabi/newpipe/player/VideoPlayer.java]: if the user already has set the auto-generated language (hence the check for "("), it is not useful to look for that as the first part of the if condition would match the caption language directly.

Comment (TobiGr) [reply app/src/main/java/org/schabi/newpipe/player/VideoPlayer.java]: Thanks

File: test.diff [hunk 3]
```diff
@@ -0,0 +1,20 @@
+diff --git a/app/src/main/java/org/schabi/newpipe/player/VideoPlayer.java b/app/src/main/java/org/schabi/newpipe/player/VideoPlayer.java
```
Comment (TobiGr) [inline test.diff]: Huh :smile: 
The diff looks a little lost here :D

Comment (Redirion) [reply test.diff]: ssht. Nothing to see here :D but you might want to check #2569 as another nice little addition for Android9+ devices

Comment (TobiGr) [reply test.diff]: I didn't see anything :D
I'll take a look at #2569 soon. There are so many open PRs at the moment... I need more time and help to review them all

Comment (TobiGr) [reply test.diff]: PS: you can squash the commits and nobody will ever know about that file :)

**Pull request merged**